### DISCUSSION TITLE
Centroid mean or ellipses in RDA plot

### DISCUSSION THREAD

**Discussion by amrvx4** [ChiLiubio/microeco#105] [Q&A] [answered]
Hi Liu, 

I was wondering if you could help me with my question. I am trying to make a RDA plot (see below). What I want to achieve is to put a big circle at the mean of each group (Ambient and Rainout shelter). Right now we can see that all the data points are color coded. But is there any way I can only place the data points at mean of the centroids? Or if I can put the ellipses around it (such as highlighted circle around lets say 70% of the data points of that group)? What I am trying to do is to make it visually better to see the differences between two groups, because right now as you can see all the data points are overlapped.

This is the code I ran to produce graph (after running cal_ordination)
KINb$plot_ordination(plot_color = "Treatment")

But I am not able to add anything else into it such as stat_ellipse etc.

![Rplot17](https://user-images.githubusercontent.com/101592620/166132104-1a9bfaa2-bdbc-48d9-a9d8-408a1e22417d.jpeg)

This is what I am trying to achieve. See example graphs below. But I am not able to apply stat_ellipse to my RDA plot using microeco package.

![ordination_unc7](https://user-images.githubusercontent.com/101592620/166132246-a4e9b427-5e06-411e-9132-384de0363085.png)
![ordination_unc8 1](https://user-images.githubusercontent.com/101592620/166132248-7b88cb5d-7ed6-4e69-9425-e89fc73ba826.png)

**Comment by ChiLiubio** [comment]
Hi @amrvx4 
Thanks for your suggestion. Actually, in the developing v0.8.0, there is a parameter plot_group_add in plot_ordination function. It is used to add ellipse or convex hull as you show. So you can first try to install the package v0.8.0 from github. The cran version will be released in the coming several days. I will also consider adding centroids and lines as you show in the first graph. You need to see the help document of the parameters as the tutorial is applied to the v0.7.5 and will be updated later. Please feel free to tell me if something is not clear or you have better suggestions on this.

Chi

**Reply by amrvx4** [reply]
Thank you for your prompt response. I updated the package and still cannot use plot_group_add parameter in plot_ordination function, it says Unused argument. Package version is 0.8

Also, I could not find documentation for 0.8 version.

Thanks,

**Comment by ChiLiubio** [comment, marked as answer]
Please manually delete the microeco package under R install directory (and find it under library directory) and  use `devtools::install_github("ChiLiubio/microeco")` to reinstall the package again. I update it just now and add the the centroid line plot as you show. Here are my testing steps:
```
library(microeco)
data(dataset)
data(env_data_16S)
t1 <- trans_env$new(dataset = dataset, add_data = env_data_16S[, 4:11])
t1$cal_ordination(method = "RDA", taxa_level = "Genus")
t1$trans_ordination(adjust_arrow_length = TRUE, max_perc_env = 1.5, max_perc_tax = 1.5, min_perc_env = 0.2, min_perc_tax = 0.2)
t1$plot_ordination(plot_color = "Group")
# centroid line plot
t1$plot_ordination(plot_type = c("point", "centroid"), plot_color = "Group", centroid_segment_linetype = 1)
# add ellipse
t1$plot_ordination(plot_color = "Group", plot_shape = "Group", plot_type = c("point", "ellipse"))
# convex hull
t1$plot_ordination(plot_color = "Group", plot_shape = "Group", plot_type = c("point", "chull"))
```
Use help(trans_env) to see the document.

Chi

**Comment by ChiLiubio** [comment]
Hi @amrvx4 
The v0.8.0 has been released on CRAN. The tutorial for v0.8 is also updated (https://chiliubio.github.io/microeco_tutorial/explainable-class.html#trans_env-class). Thanks for your suggestion! 

Best
Chi

**Reply by amrvx4** [reply]
Great. Thank you so much for putting your time in this.
